File: to-chat-spring-boot-sample-java-app/src/main/java/de/larmic/sample/bot/SampleApplication.java
package de.larmic.sample.bot;

import chat.to.server.bot.MessageSender;
import chat.to.server.bot.autoconfigure.MessageReceiver;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.Bean;

@SpringBootApplication
public class SampleApplication {

    private final Logger log = LoggerFactory.getILoggerFactory().getLogger(this.getClass().getName());

    public static void main(String[] args) {
        SpringApplication.run(SampleApplication.class, args);
    }

    @Bean
    public MessageReceiver receiver(final MessageSender messageExchangeService) {
        return message -> {
            log.info("Message '{}' from '{}' received", message.getText(), message.getCreateDate());
            messageExchangeService.sendMessage(message.getLinks().getChannel().getIdentifier(), "Hi there! Message '${message.text}' received! ¯\\_(ツ)_/¯");
        };
    }
}

File: to-chat-spring-boot-sample-kotlin-app/src/main/kotlin/de/larmic/sample/bot/SampleApplication.kt
package de.larmic.sample.bot

import chat.to.server.bot.Message
import chat.to.server.bot.MessageSender
import chat.to.server.bot.autoconfigure.MessageReceiver
import org.slf4j.LoggerFactory
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.context.annotation.Bean

@SpringBootApplication
class GreeterSampleApplication {

    private val log = LoggerFactory.getLogger(this::class.java)

    @Bean
    fun receiver(messageSender: MessageSender) = object : MessageReceiver {
        override fun receiveMessage(message: Message) {
            log.info("Message '${message.text}' from '${message.createDate}' received")
            messageSender.sendMessage(message.links.channel.identifier, "Hi there! Message '${message.text}' received!")
        }
    }
}

fun main(args: Array<String>) {
    runApplication<GreeterSampleApplication>(*args)
}